client/src/components/MapView.vue

```diff
@@ -34,7 +72,56 @@ export default {
     }
   },
   methods: {
-    async createMap(waterBoundaryData) {
+    mapOfObject(object) {
+      return new Map(Object.entries(object));
+    },
+
+    /**
+     * Creates popup component at the given lngLat.
+     *
+     * Passes propsData to the MapPopupContent component which is nested in the
+     * popup.
+     */
+    createMapPopup(lngLat, popupData) {
+      new mapbox.Popup(
+          {className: 'mapbox-popup', maxWidth: 258, maxHeight: 256})
+          .setLngLat(lngLat)
+          .setHTML(POPUP_CONTENT_BASE_HTML) // Add basic div to mount to.
+          .addTo(this.map);
+
+      // Creates component which extends MapPopupContent and mounts it onto the
+      // div passed to the Popup.
+      const MapPopup = defineComponent({
+        extends: MapPopupContent,
+      })
+      nextTick(() => {
+        createApp(MapPopup, popupData).mount(`#${POPUP_CONTENT_BASE_ID}`)
+      });
+    },
+
+    /**
+     * Sets up interaction handlers for map.
+     */
+    setUpInteractionHandlers() {
+      this.map.on('click', 'epa-violations-population-style', async (e) => {
```

Comment:
Can we use Vue's [`@click`](https://vuejs.org/guide/essentials/event-handling.html#listening-to-events) here instead of defining a new listener? If not, can you add a comment why?